packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -284,55 +316,116 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
     }
   }
 
-  bool _shouldThrow(int statusCode) {
-    return statusCode < 200 || statusCode > 299;
-  }
-
-  RestResponse _formatRestResponse(Map<String, dynamic> res) {
+  AWSStreamedHttpResponse _formatRestResponse(Map<String, dynamic> res) {
     final statusCode = res['statusCode'] as int;
     final headers = res['headers'] as Map?;
```

Comment:
callout: realized I forgot to pass headers to response here, will fix in next rev

Reply:
done